File: .config/ags/js/controlCenter/widgets/Clock.js
import GLib from "gi://GLib";
import Widget from "resource:///com/github/Aylur/ags/widget.js";

export default ({ format = "%H:%M", interval = 1000 } = {}) =>
  Widget.Box({
    hexpand: true,
    vpack: "center",
    hpack: "center",
    children: [
      Widget.Label({
        class_name: "clock__label",
        connections: [
          [
            interval,
            (label) =>
              (label.label = GLib.DateTime.new_now_local().format(format)),
          ],
        ],
      }),
    ],
  });

File: .config/astal/lua/widget/dashboard/clock.lua
local widget = function()
	local time = Variable(""):poll(1000, function()
		return GLib.DateTime.new_now_local():format("%H:%M - %A %e.")
	end)

	return Widget.Box({
		class_name = "element",
		Widget.Label({
			class_name = "clock",
			on_destroy = function()
				time:drop()
			end,
			label = time(),
		}),
	})
end

return widget
